packages/conf-bmake/conf-bmake.1.0/files/detect_program.sh

```diff
@@ -0,0 +1,28 @@
+### detect_program.sh -- Detect program and save its path
+
+detect_which()
+{
+    ( IFS=':'
+      for path_elt in ${PATH}; do
+          if [ -x "${path_elt}/$1" ]; then
+              printf '%s' "${path_elt}/$1"
+              exit 0
+          fi
+      done
+
+      printf 'false'
+      exit 1 )
+}
+
+detect_config()
+{
+    local program_path
+
+    program_path=$(detect_which "$1")
+    cat <<EOF
+path: "${program_path}"
+EOF
+}
+
+mkdir -p "$1"
+detect_config "$2" > "$1/opam.config"
```

Comment:
I am not sure I understand this properly. This opam package wants to detect at build time the presence of a BSD Make compatible program. If such a program is found, its path should be saved somewhere so that other packages can refer it as `%{conf-bmake:path}%`.

@samoht I am puzzled by the suggestion of writing the generated configuration file to the `files/` directory because the build should not write to the directory holding files used by opam. Could you please clarify this?